vlib/v/checker/tests/struct_scope_err.vv

```diff
@@ -0,0 +1,18 @@
+fn main() {
+
+	struct Foobar {
+		foo int
+		bar int
+	}
+
+	if true {
+		fb := Foobar{5, 6}
+		println(fb.foo)
+		x()
+	}
+}
+
+fn x() {
+	fb := Foobar{7, 8}
+	println(fb.bar)
+}
```

Comment:
Can you have `fn g() {` that has a *different* `struct Foobar {` declared in it?
Something like `struct Foobar { name string }` for example?